src/dotnet/APIView/APIViewWeb/Managers/PullRequestManager.cs

```diff
@@ -248,13 +248,20 @@ private async Task ClosePullRequestAPIRevision(PullRequestModel pullRequestModel
         }
 
         private async Task CreateAPIRevisionIfRequired(CodeFile codeFile, string originalFileName, MemoryStream memoryStream,
-            PullRequestModel pullRequestModel, CodeFile baselineCodeFile, MemoryStream baseLineStream, string baselineFileName, CreateAPIRevisionAPIResponse responseContent)
+            PullRequestModel pullRequestModel, CodeFile baselineCodeFile, MemoryStream baseLineStream, string baselineFileName, CreateAPIRevisionAPIResponse responseContent, string packageType = null)
         {
             // fetch review for the package or create brand new review
             var review = await _reviewManager.GetReviewAsync(language: codeFile.Language, packageName: codeFile.PackageName);
             if (review == null)
             {
-                review = await _reviewManager.CreateReviewAsync(language: codeFile.Language, packageName: codeFile.PackageName, isClosed: false);
+                // Parse package type if provided, otherwise pass null for automatic classification
+                APIViewWeb.Helpers.PackageType? parsedPackageType = null;
```

Comment:
Should this be a helper, or just a model? (do whatever is consistent with existing code-I have no preference)